src/cmd/support_archive/tarball.go

```diff
@@ -0,0 +1,96 @@
+package support_archive
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"strings"
+	"time"
+)
+
+const tarFileName = "%s/operator-support-archive-%s.tgz"
+const defaultTargetDir = "/tmp/dynatrace-operator"
+
+type tarball struct {
+	tarWriter  *tar.Writer
+	gzipWriter *gzip.Writer
+	tarFile    *os.File
+}
+
+func newTarball(ctx *supportArchiveContext) (*tarball, error) {
+	var err error
+	tarball := tarball{}
+	if tarball.tarFile, err = selectAndCreateTargetFile(ctx); err != nil {
+		return nil, err
+	}
+	tarball.gzipWriter = gzip.NewWriter(tarball.tarFile)
+	tarball.tarWriter = tar.NewWriter(tarball.gzipWriter)
+	return &tarball, nil
+}
+
+func (t *tarball) close() {
+	if t.tarWriter != nil {
+		t.tarWriter.Close()
+	}
+	if t.gzipWriter != nil {
+		t.gzipWriter.Close()
+	}
+	if t.tarFile != nil {
+		t.tarFile.Close()
+	}
+}
+
+func (t *tarball) addFile(fileName string, file io.Reader) error {
+	buffer := &bytes.Buffer{}
+	io.Copy(buffer, file)
+
+	header := &tar.Header{
+		Name: fileName,
+		Size: int64(buffer.Len()),
+		Mode: int64(fs.ModePerm),
+	}
+
+	err := t.tarWriter.WriteHeader(header)
+	if err != nil {
+		return fmt.Errorf("could not write header for file '%s', got error '%w'", fileName, err)
```

Comment:
pls use `errors` package (`errors.WithStack` or `errors.WithMessagef` or something like that :) )